// SPDX-FileCopyrightText: 2025 Carnegie Mellon University - LivingEdgeLab
//
// SPDX-License-Identifier: GPL-2.0-only

syntax = "proto3";
package protocol.report_service;

import "common.proto";
import "google/protobuf/any.proto";

/*
 * Used to report messages to the Swarm Controller server
 * or to other collaborative vehicles
 */
service Report {
  // Send a report to the server, or to an intended recipient
  rpc SendReport (ReportRequest) returns (ReportResponse) {}
}

message ReportMessage {
  int32 report_code = 1; // Integer report code, interpreted by the backend
  google.protobuf.Any report_data = 2; // Extra data about report
}

message ReportRequest {
  protocol.common.Request request = 1;
  bytes identity = 2; // UTF-8 encoded identity of sender
  bytes recipient = 3; // UTF-8 encoded identity of recipient
  ReportMessage report = 4;
}

message ReportResponse {
  protocol.common.Response response = 1;
  bytes identity = 2; // UTF-8 encoded identity of responder
}
